tools/detekt/src/main/kotlin/com/datadog/tools/detekt/rules/sdk/ThreadSafety.kt

```diff
@@ -59,7 +69,7 @@ class ThreadSafety : Rule() {
         }
     }
 
-    private var parentFunGroup: ThreadGroup = ThreadGroup.UNKNOWN
+    private val parentFunGroupStack: MutableList<ThreadGroup> = mutableListOf(ThreadGroup.UNKNOWN)
```

Comment:
minor: since we are treating it like a stack, we can use `Stack` type here (or `Deque` -> `LinkedList`) which will allows us simply use `push`/`pop`/`peek` methods instead of `add(index, ...)` and `removeAt(index, ...)`